src/reachability/utils/reachability.jl

```diff
@@ -27,24 +27,24 @@ end
 function _forward_network_collect(solver::Solver,
                                   nnet::Network,
                                   input,
-                                  collect_f=last,
-                                  T=Any)
+                                  collect_f=last)
     zs = (input, input)
-    collection::Vector{T} = [collect_f(zs)]
+    elem = collect_f(zs)
+    T = typeof(elem)
+    collection = T[elem]
```

Comment:
For bounds we end up with the same type b/c collect_f will always return a Hyperrectangle. If we don't assume that collect_f always leads to the same type then I'm still not sure what to do here. I tried some to get collect_f to tell us the type it will give back, but we don't know at the start what type we'll be inputting into it